tests/unit/test_part_4.py

```diff
@@ -0,0 +1,64 @@
+import random
```

Comment:
tests look much better

Reply:
let's make a call tomorrow

Reply:
by the way, good job, really like to read your code and see your professional growth.